types/coin_test.go

```diff
@@ -1276,6 +1276,40 @@ func (s *coinTestSuite) TestMarshalJSONCoins() {
 	}
 }
 
+func (s *coinTestSuite) TestCoinValidate() {
+	testCases := []struct {
+		name    string
+		coin    sdk.Coin
+		wantErr string
+	}{
+		{"nil coin: nil Amount", sdk.Coin{}, "invalid denom"},
+		{"non-blank coin, nil Amount", sdk.Coin{Denom: "atom"}, "nil Amount"},
```

Comment:
```suggestion
		{"non-blank coin, nil Amount", sdk.Coin{Denom: "atom"}, "amount is nil"},
```

Just small fix for test to pass

Reply:
Oops, thanks @JeancarloBarrios! I had somehow just reverted that.

Reply:
Done and thank you @JeancarloBarrios!